fdbclient/StorageServerInterface.h

```diff
@@ -122,7 +122,7 @@ struct StorageServerInterface {
 				getKeyValuesStream =
 				    RequestStream<struct GetKeyValuesStreamRequest>(getValue.getEndpoint().getAdjustedEndpoint(13));
 				getKeyValuesAndFlatMap =
-				    RequestStream<struct GetKeyValuesAndFlatMapRequest>(getValue.getEndpoint().getAdjustedEndpoint(17));
+				    RequestStream<struct GetKeyValuesAndFlatMapRequest>(getValue.getEndpoint().getAdjustedEndpoint(14));
```

Comment:
Why did you change it to 14?

Reply:
Because the team is concern that having placeholders is not clean and may have unknowns regarding compatibility. I'm moving it up to 14 in master too https://github.com/apple/foundationdb/pull/5955

Reply:
I wonder if we should conditionally initialize these request streams based on the protocol version. Right now we're initializing them with a remote endpoint that doesn't exist when reading a message written by 7.0 on master. What we used to do is just not initialize the request stream if it didn't exist at the version it was written. I guess in practice it doesn't cause any problems though - if it did our restart tests should turn that up.

Reply:
There's a comment above:
```
		// StorageServerInterface is persisted in the database, so changes here have to be versioned carefully!
		// To change this serialization, ProtocolVersion::ServerListValue must be updated, and downgrades need to be
		// considered
```

@sfc-gh-etschannen do you think we're doing the right thing here?

Reply:
what if a cluster has a mixed versions of fdbservers, say 6.3 and 7.0? 
If endpoint(14) means one interface in 6.3 and a different interface in 7.0, will that cause problems?

Reply:
Flatbuffers file identifier checking should mitigate it, unless the request type for the streams happens to be the same. Btw since 7.0 and 6.3 are protocol incompatible we're not too worried about network messages here - we're mainly worried about the serialized StorageServerInterface's that are written to the database